src/lib/core.d.ts

```diff
@@ -1530,7 +1530,7 @@ interface Int8Array {
       * @param thisArg If provided, it will be used as the this value for each invocation of 
       * predicate. If it is not provided, undefined is used instead.
       */
-    findIndex(predicate: (value: number) => boolean, thisArg?: any): number;
+    findIndex(predicate: (value: number) => boolean, thisArg?: any): number | undefined;

```

Comment:
@mhegazy And should I move the other commits after the first one into a separate PR as well?